### PR TITLE
feat: update `gh` spec

### PR DESCRIPTION
**Pull request by nicholasoxford**
**What kind of change does this PR introduce? (Bug fix, feature, docs update, ...)**

Updated the GH CLI autocomplete to include GH 2.0 commands

**What is the current behavior? (You can also link to an open issue here)**

issing

**What is the new behavior (if this is a feature change)?**

Adds sections like extensions and ssh-key

**Additional info:**

https://cli.github.com/manual/

Sorry for the amount of commit messages

### CODE REVIEW

**Review by fedeci — COMMENTED** [review]
Can you run `npm run lint:fix && npm run test` locally and then push again?

**Comment by nicholasoxford** [comment]
I ran it, but it seemed to not stage any changes.

Screen Shot 2021-09-14 at 10 16 34 AM

File: dev/gh.ts [hunk 1]
```diff
@@ -1146,6 +1205,347 @@ const completionSpec: Fig.Spec = {
         },
       ],
     },
+    {
+      name: "run",
+      description: "View details about workflow runs",
+      options: [ghOptions.ghRunOptions],
```
Comment (fedeci) [inline dev/gh.ts:1211]: The fact that `tsc` is not marking this as an error is a problem🤔

Comment (nicholasoxford) [reply dev/gh.ts:1211]: What is the type collision?

File: dev/gh.ts [hunk 2]
```diff
@@ -1146,6 +1205,347 @@ const completionSpec: Fig.Spec = {
...
+      subcommands: [
+        {
+          name: "download",
+          description: "Download artifacts generated by a workflow run",
+          options: [
+            ghOptions.ghRunOptions,
```
Comment (fedeci) [inline dev/gh.ts:1217]: I don't think it needs to be repeated in the subcommands too.

File: dev/gh.ts [hunk 3]
```diff
@@ -35,6 +35,21 @@ const ghGenerators: Record<string, Fig.Generator> = {
 const ghOptions: Record<string, Fig.Option> = {
   help: { name: "--help", description: "Show help for command" },
 };
+const ghUpgradeOptions: Record<string, Fig.Option> = {
+  help: { name: "--help", description: "Show help for command" },
+  all: { name: "--all", description: "Upgrade all extensions" },
+  force: { name: "--force", description: "Force upgrade extensions" },
```
Comment (fedeci) [inline dev/gh.ts:41]: I don't find it logically correct to make this a dictionary. An array is enough. Same for `ghRunOptions`

**Review by fedeci — CHANGES_REQUESTED** [review]

File: dev/gh.ts (hunk 1 above)
Comment (fedeci) [reply dev/gh.ts:1211]: The fact that `ghOptions` does not contain a `ghRunOptions` prop.

File: dev/gh.ts (hunk 3 above)
Comment (nicholasoxford) [reply dev/gh.ts:41]: Out of curiosity, what's the logic behind making a dictionary?

Comment (nicholasoxford) [reply dev/gh.ts]: Honestly I can't believe I put ghOptins.ghRunOptions 🤣, I probably only put it because it wasn't failing.

**Comment by QuiiBz** [comment]
Hi @nicholasoxford, thanks a lot for this PR! Could you rebase your branch to keep a clean history?

**Comment by nicholasoxford** [comment]
rebase did not go as originally planned.

Reopened here: https://github.com/withfig/autocomplete/pull/589

**Pull request closed without merging**